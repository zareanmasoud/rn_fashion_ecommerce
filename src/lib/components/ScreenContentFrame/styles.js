import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  imgBackground: {
    width: '100%',
    height: undefined,
    aspectRatio: 1,
    resizeMode: 'contain',
    position: 'absolute',
    bottom: 0,
  },
  safeAreaView: {
    flex: 1,
  },
  contentContainer: {
    flexGrow: 1,
  },
});

export default styles;
