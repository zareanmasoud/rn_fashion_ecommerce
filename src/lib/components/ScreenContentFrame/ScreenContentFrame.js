import React, {Component} from 'react';
import {SafeAreaView, ImageBackground, ScrollView, StatusBar} from 'react-native';
import styles from './styles';
import R from "res/R";

export class ScreenContentFrame extends Component {
  render() {
    return (
      <>
        <StatusBar barStyle="dark-content" backgroundColor={R.colors.title}/>
        {/*<ImageBackground source={R.images.main_background} style={styles.imgBackground}/> // or view with backgroundColor */}
        <SafeAreaView style={styles.safeAreaView}>
          <ScrollView contentContainerStyle={styles.contentContainer}>
            {
              this.props.children
            }
          </ScrollView>
        </SafeAreaView>
      </>
    );
  }
}
