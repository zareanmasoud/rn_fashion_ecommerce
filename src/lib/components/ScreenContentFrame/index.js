import ScreenContentFrame from './ScreenContentFrame';

export {ScreenContentFrame as default};
