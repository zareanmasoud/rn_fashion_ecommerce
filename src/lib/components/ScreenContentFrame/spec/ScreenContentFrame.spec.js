import React from 'react';
import {View} from 'react-native';
import {shallow} from 'enzyme';
import ScreenContentFrame from '../ScreenContentFrame';

describe('ScreenContentFrame Component', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const wrapper = shallow(
        <ScreenContentFrame>
          <View />
        </ScreenContentFrame>,
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
