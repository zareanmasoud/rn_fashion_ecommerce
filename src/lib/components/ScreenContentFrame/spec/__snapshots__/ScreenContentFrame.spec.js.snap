// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenContentFrame Component Rendering should match to snapshot 1`] = `
<SafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <StatusBar
    animated={false}
    backgroundColor="#ffffff"
    barStyle="dark-content"
    showHideTransition="fade"
  />
  <View />
</SafeAreaView>
`;
